crypto.go

```diff
@@ -718,6 +813,12 @@ func (s hkdfScheme) ID() KDFID {
 	panic(fmt.Sprintf("Unsupported hash: %d", s.hash))
 }
 
+func (s hkdfScheme) kemID() []byte {
+	idBuffer := make([]byte, 2)
+	binary.BigEndian.PutUint16(idBuffer, uint16(s.ID()))
```

Comment:
This method doesn't make sense. `kemID` shouldn't be part of `hkdfScheme`. I believe this outputs the KDF ID, which would explain why DeriveKeyPair is failing in my KAT

Reply:
Hah, yeah, good point. The intent of putting it on the KDF interface was to say, "give me the suite ID when I'm being used as part of a KEM," but as you point out that's somewhat confusing (and it was using the wrong ID)! I moved things around to address this. Now, each time the implicit suite_id is required, the calling code passes the KEM ID to a helper function to compute this.